subworkflows/local/prepare_genome.nf

```diff
@@ -0,0 +1,72 @@
+//
+// Prepare reference genome files
+//
+
+include { BISMARK_GENOMEPREPARATION   } from '../../modules/nf-core/bismark/genomepreparation/main'
+include { BWAMETH_INDEX               } from '../../modules/nf-core/bwameth/index/main'
+include { SAMTOOLS_FAIDX              } from '../../modules/nf-core/samtools/faidx/main'
+
+workflow PREPARE_GENOME {
+
+    main:
+    versions = Channel.empty()
+
+    fasta = Channel.empty()
+    bismark_index = Channel.empty()
+    bwameth_index = Channel.empty()
+    fasta_index = Channel.empty()
+
+    // FASTA, if supplied
+    if (params.fasta) {
+        fasta = file(params.fasta)
+    }
+
+    // Aligner: bismark or bismark_hisat
+    if( params.aligner =~ /bismark/ ){
```

Comment:
I like the idea of avoiding references to params in subworkflows (referencing a param forces the re-user of the workflow to add that param, whereas if we just use a channel, then can choose to hard-code the value or make it a param, or make it a param under a different name, etc.) Should this also be moved to an input 'param map'?

Reply:
I had this same conversation with @drpatelh - I started moving the params out of this subworkflow but it made the logic in the main workflow so complex that it basically wasn't worth having it as a subworkflow any longer.

So generally yes, but here I made an exception (@drpatelh set a precedent of the same in @nf-core /rnaseq).